GraphConsoleAppV3/GraphRequests.cs

```diff
@@ -0,0 +1,1534 @@
+
+            // Search for a group and assign the newUser to the found group
+
+            //*********************************************************************
+            // Search for a group using a startsWith filter (displayName property)
+            //*********************************************************************
+            Group retrievedGroup = new Group();
+            Console.WriteLine("\nSearch for a group to add the current user to:");
+            string groupName = Console.ReadLine();
+
+            List<IGroup> foundGroups = null;
+            try
+            {
+                IPagedCollection<IGroup> groupsCollection = await client.Groups
+                    .Where(group => group.DisplayName.StartsWith(groupName))
+                    .ExecuteAsync();
+                foundGroups = groupsCollection.CurrentPage.ToList();
+            }
+            catch (Exception e)
+            {
+                Program.WriteError("\nError getting Group {0} {1}",
+                    e.Message, e.InnerException != null ? e.InnerException.Message : "");
+            }
+            if (foundGroups != null && foundGroups.Count > 0)
+            {            
+                int pickedGroupIndex = default(int);
+
+                if (foundGroups.Count == 1)
+                {
+                    pickedGroupIndex = 0;
+                }
+                else
+                {
+                    for (int i = 0; i < foundGroups.Count; i++)
+                    {
+                        Console.WriteLine("\n{0}. {1}", i + 1, foundGroups[i].DisplayName);
+                    }
+
+                    string keyString;
+                    int key;
+                    do
+                    {
+                        Console.WriteLine("Pick the group you want to add the new user to by entering a number:");
+                        keyString = Console.ReadLine();
+                    } while (!(int.TryParse(keyString, out key) && key > 0 && key <= foundGroups.Count));
+                    pickedGroupIndex = key - 1;
+                }
+                // add new user to picked group
+                if (foundGroups[pickedGroupIndex].ObjectId != null)
+                {
+                    try
+                    {
+                        retrievedGroup = (Group)foundGroups[pickedGroupIndex];
+                        retrievedGroup.Members.Add(newUser as DirectoryObject);
+                        await retrievedGroup.UpdateAsync();
+                    }
+                    catch (Exception e)
+                    {
+                        Program.WriteError("\nError assigning member to group. {0} {1}",
+                            e.Message, e.InnerException != null ? e.InnerException.Message : "");
+                    }
+                }
+            }
+            else
+            {
+                Console.WriteLine("Group Not Found based on search criteria, and hence user not added to group");
```

Comment:
Still see "Group not Found based on search"

---
In reply to: [80803034](https://github.com/Azure-Samples/active-directory-dotnet-graphapi-console/pull/37#discussion_r80803034) [](ancestors = 80803034,80545283)